### PR TITLE
Documentation and dispatching change event

### PR DESCRIPTION
**Pull request by krshnpatel-d2l**
Changes made in this PR:
~ Added documentation for `input-number` component and links to it
~ Minor typo fix in `input-text.md`
~ Dispatching change event when value is changed
~ Added unit tests for dispatching change event

### CODE REVIEW

File: components/inputs/README.md [hunk 1]
```diff
@@ -94,7 +95,6 @@ Alternately, the `<d2l-input-fieldset>` component can accomplish this for you. T
 ## Future Enhancements
 
 - Date & time range with validation
-- Numeric input
```
Comment (dlockhart) [inline components/inputs/README.md:97]: Good catch!

File: components/inputs/docs/input-number.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
+# Number Inputs
+
+The `<d2l-input-number>` element is a wrapper around the `<d2l-input-text>` component. It's intended for inputting numbers only.
```
Comment (dlockhart) [inline components/inputs/docs/input-number.md:3]: We may choose one day to not simply wrap `d2l-input-text`, so I don't think we need to leak this implementation details. You could say something like:
> The `<d2l-input-number>` element is similar to `<d2l-input-text>`, except it's intended for inputting numbers only.

File: components/inputs/docs/input-number.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  label="Label"
+  value="0">
+</d2l-input-number>
+```
+
+**Properties:**
+
+| Property | Type | Description |
+|--|--|--|
+| `label` | String, required | Label for the input. |
```
Comment (dlockhart) [inline components/inputs/docs/input-number.md:21]: These seem to be alphabetical except for `label`

File: components/inputs/docs/input-text.md [hunk 4]
```diff
@@ -24,7 +24,7 @@ The `<d2l-input-text>` element is a simple wrapper around the native `<input typ
 | `autocomplete` | String | Specifies which types of values [can be autofilled](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete) by the browser |
 | `autofocus` | Boolean | When set, will automatically place focus on the input |
 | `disabled` | Boolean | Disables the input |
-| `label-hidden` | Boolean | Hides the labe visually (moves it to the input's `aria-label` attribute) |
+| `label-hidden` | Boolean | Hides the label visually (moves it to the input's `aria-label` attribute) |
```
Comment (dlockhart) [inline components/inputs/docs/input-text.md:27]: 👍

File: components/inputs/input-number.js [hunk 5]
```diff
@@ -147,6 +147,11 @@ class InputNumber extends FormElementMixin(LocalizeCoreElement(LitElement)) {
 		this._formattedValue = value;
 		await this.updateComplete;
 		this.value = parseNumber(value);
+
+		this.dispatchEvent(new CustomEvent(
+			'change',
+			{ bubbles: true, composed: false }
+		));
```
Comment (dlockhart) [inline components/inputs/input-number.js:154]: Could you add some tests for this as well? We want to make sure that if the underlying value didn't actually change (even though the value the user typed in might have), that we don't fire the event.

File: components/inputs/docs/input-number.md (hunk 3 above)
Comment (krshnpatel-d2l) [reply components/inputs/docs/input-number.md:21]: This is because it's a `required` attribute. I see that `label` is always first for other docs as well.

Comment (dlockhart) [reply components/inputs/docs/input-number.md:21]: Ah ok cool 👍

**Review by dlockhart — APPROVED** [review]
Looks good!

**Pull request merged**